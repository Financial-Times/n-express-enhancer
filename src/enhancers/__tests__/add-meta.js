import addMeta from '../add-meta';

describe('addMeta', () => {
	it('create enhancer that can add meta to action functions both async/non-async', async () => {
		const actionFunctionBundle = {
			methodA: (param, meta) => meta,
			methodB: async (param, meta) => meta,
		};
		const addingMeta = { service: 'some-service' };
		const enhanced = addMeta(addingMeta)(actionFunctionBundle);
		const resultA = enhanced.methodA({});
		const resultB = await enhanced.methodB({});
		expect(resultA).toMatchSnapshot();
		expect(resultB).toMatchSnapshot();
	});

	it('catches errors thrown from action function both async/non-async', async () => {
		const errorInstance = { status: 404, meesage: 'some error message' };
		const actionFunction = () => {
			throw errorInstance;
		};
		const asyncActionFunction = async () => {
			throw await errorInstance;
		};
		const addServiceMeta = addMeta({ service: 'some-service' });
		try {
			addServiceMeta(actionFunction)();
		} catch (e) {
			expect(e).toEqual(errorInstance);
		}
		try {
			await addServiceMeta(asyncActionFunction)();
		} catch (e) {
			expect(e).toEqual(errorInstance);
		}
	});

	it('throws error if input of addMeta is not an object, e.g. in case the meta is forgotten', () => {
		const operationFunction = () => {};
		expect(() => addMeta('')).toThrowErrorMatchingSnapshot();
		expect(() => addMeta(operationFunction)).toThrowErrorMatchingSnapshot();
	});
});
