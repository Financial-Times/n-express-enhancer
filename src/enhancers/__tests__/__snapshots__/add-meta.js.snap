// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`addMeta create enhancer that can add meta to action functions both async/non-async 1`] = `
Object {
  "service": "some-service",
}
`;

exports[`addMeta create enhancer that can add meta to action functions both async/non-async 2`] = `
Object {
  "service": "some-service",
}
`;

exports[`addMeta throws error if input of addMeta is not an object, e.g. in case the meta is forgotten 1`] = `"input of addMeta needs to be an object"`;

exports[`addMeta throws error if input of addMeta is not an object, e.g. in case the meta is forgotten 2`] = `"input of addMeta needs to be an object"`;
